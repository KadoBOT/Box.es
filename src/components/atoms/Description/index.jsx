import styled from 'styled-components';

const Description = styled.div`
  font-size: 16px;
  color: gray;
  max-height: ${props => props.complete ? '190px' : '80px'};
  overflow: hidden;
  line-height: 20px;
  display: ${props => props.complete ? 'block' : '-webkit-box'};
  overflow: ${props => props.complete ? 'auto' : 'hidden'};
  -webkit-box-orient: vertical;
  -webkit-line-clamp: ${props => props.comment ? '2' : '3'};;
  line-height: 20px;
  position: relative;
  word-break: break-word;
  -webkit-box-shadow: ${props => props.complete && 'inset 0px -30px 22px -17px rgba(0,0,0,0.12)'};
  -moz-box-shadow: ${props => props.complete && 'inset 0px -30px 22px -17px rgba(0,0,0,0.12)'};
  box-shadow: ${props => props.complete && 'inset 0px -30px 22px -17px rgba(0,0,0,0.12)'};
  padding-bottom: ${props => props.complete && '25px'};
`

export default Description
