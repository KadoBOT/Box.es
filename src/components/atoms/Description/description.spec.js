import React from 'react'
import renderer from 'react-test-renderer';
import Description from './index'

it('should render the Description', () => {
    const wrapper = renderer.create(
        <Description />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
});

it('should render the Complete Description', () => {
    const wrapper = renderer.create(
        <Description complete/>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
});
