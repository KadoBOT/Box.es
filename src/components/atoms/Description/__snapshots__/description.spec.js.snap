exports[`test should render the Complete Description 1`] = `
<div
  className="bzxTKv" />
`;

exports[`test should render the Description 1`] = `
<div
  className="cabTZo" />
`;
